sbncode/CAFMaker/CAFMaker_module.cc

```diff
@@ -1280,11 +1278,16 @@ void CAFMaker::produce(art::Event& evt) noexcept {
         thisShower = fmShower.at(iPart);
       }
 
-      if (!thisTrack.empty())  { // it's a track!
+      assert(thisParticle.size() == 1);
+     
+      SRPFP pfp;
+      // ...
+      rec.reco.pfp.push_back(std::move(pfp));
+      const larpandoraobj::PFParticleMetadata *pfpMeta = (fmPFPMeta.at(iPart).empty()) ? NULL : fmPFPMeta.at(iPart).at(0).get();
+      FillPFPVars(thisParticle, primary, pfpMeta, rec.reco.pfp.back());
+
+      if (!thisTrack.empty())  { // it has a track!
```

Comment:
Sorry, @rhowell42, there was a misunderstanding... that `// ...` was meant to represent all the code below.
Let me see if I can get it right... but it will be a lot of boxes.
```suggestion
      SRPFP pfp;
      const larpandoraobj::PFParticleMetadata *pfpMeta = (fmPFPMeta.at(iPart).empty()) ? NULL : fmPFPMeta.at(iPart).at(0).get();
      FillPFPVars(thisParticle, primary, pfpMeta, pfp);

      if (!thisTrack.empty())  { // it has a track!
```